### ISSUE TITLE
[Feature Request]:

### ISSUE THREAD

**Issue by davloperez** [storybookjs/storybook#19576]
### Is your feature request related to a problem? Please describe

There is no way to use a nullable Date prop.

Example:
```tsx
type MyComponentProps = {
    myDate: Date | null;
};
const MyComponent = (props: MyComponentProps) =>
    (<span>{props.myDate?.toString() ?? 'No date' }</span>);
```

This component prop `myDate` can not be represented with current Controls addon, as date is converted to a numeric timestamp (noted in #11822, with workaround), **but also, the null value can not be represented with the current Control for date.**

The control for this date prop always shows a date (1970-01-01, 00:00), even after using `useArgs` hook and set explicit `null` value to `myDate` prop.

### Describe the solution you'd like

The solution could be just to let the date Control to be empty, which will represent a null value. Or maybe show a toggle button to set null or a mandatory Date.

### Describe alternatives you've considered

If it breaks existing usages of date Controls, it could be opt-in activated via control options in parameters or something like that.

### Are you able to assist to bring the feature to reality?

no

### Additional context

![image](https://user-images.githubusercontent.com/1970817/197266716-efe7484a-7724-409e-9c5e-5bc5968a737d.png)

**Comment by life4aiur** [comment]
The root cause of this issue seems to be the same as the one I reported here: #19215 

TL;DR - Most storybook controls can't be unset

**Comment by vardumper** [comment]
I ran into the same situation not wanting to set a Date always. Copilot came up with the options setting on the control which works as well.
 
```
lastModifiedStartDate: {
    control: { type: "date", options: [null] },
},
```

**Comment by WesselKroos** [comment]
It would be helpful if we could reset "nullable" properties to undefined per control. Some stories have a prefilled value from the args parameter, but can never be reset to undefined.

(We are using typescript, so our properties are derived from typed properties of which some are optional.)

**Comment by shilman** [comment]
@WesselKroos if your story sets an arg to `undefined` you can reset the controls for the story in the UI. It's not ideal but it's something.
